compiler/rustc_middle/src/ty/print/pretty.rs

```diff
@@ -909,12 +909,25 @@ pub trait PrettyPrinter<'tcx>:
                 }
 
                 for (assoc_item_def_id, term) in assoc_items {
-                    // Skip printing `<[generator@] as Generator<_>>::Return` from async blocks
-                    if let Some(ty) = term.skip_binder().ty() &&
-                       let ty::Projection(ty::ProjectionTy { item_def_id, .. }) = ty.kind() &&
-                       Some(*item_def_id) == self.tcx().lang_items().generator_return() {
-                        continue;
-                    }
+                    // Skip printing `<[generator@] as Generator<_>>::Return` from async blocks,
+                    // unless we can find out what generator return type it comes from.
```

Comment:
Seems to say the inverse of what it should say

```suggestion
                    // Skip printing `<[generator@] as Generator<_>>::Return` from async blocks
                    // if we can find out what generator return type it comes from.
```

Reply:
Maybe our perspectives are inverted -- so it does two things, 1. skips printing `<[generator@] as Generator<_>>::Return` _regardless_, but will instead print `Output=Type` if it can find the return type. If it can't, it skips printing altogether.

Reply:
Oh, then the comment makes sense 👍🏼